bft_interp/src/lib.rs

```diff
@@ -1,39 +1,510 @@
+        let result = vm.move_head_right();
+
+        assert!(result.is_ok());
+        assert_eq!(vm.head, 1000);
+        assert_eq!(vm.cells.capacity(), 2000)
+    }
+
+    // For u8, does incrementing without wrapping work?
+    #[test]
+    fn test_u8_increment_no_wrap() {
+        let test_program = test_program();
+        let mut vm: VirtualMachine<u8> = VirtualMachine::new(&test_program, None, false);
+
+        vm.cells[0] = 10;
+        vm.increment_cell();
+
+        assert_eq!(vm.cells[0], 11);
+    }
+
+    // For u8, does incrementing wrap around the max value?
+    #[test]
+    fn test_u8_increment_wrap() {
+        let test_program = test_program();
+        let mut vm: VirtualMachine<u8> = VirtualMachine::new(&test_program, None, false);
+
+        vm.cells[0] = u8::MAX;
+        vm.increment_cell();
+
+        assert_eq!(vm.cells[0], u8::MIN);
+    }
+    // For u8, does decrementing without wrapping work?
+    #[test]
+    fn test_u8_decrement_no_wrap() {
+        let test_program = test_program();
+        let mut vm: VirtualMachine<u8> = VirtualMachine::new(&test_program, None, false);
+
+        vm.cells[0] = 10;
+        vm.decrement_cell();
+
+        assert_eq!(vm.cells[0], 9);
+    }
+
+    // For u8, does decrementing wrap around the min value?
+    #[test]
+    fn test_u8_decrement_wrap() {
+        let test_program = test_program();
+        let mut vm: VirtualMachine<u8> = VirtualMachine::new(&test_program, None, false);
+
+        vm.cells[0] = u8::MIN;
+        vm.decrement_cell();
+
+        assert_eq!(vm.cells[0], u8::MAX);
+    }
+
+    // does reading a byte into a cell work?
+    #[test]
+    fn test_read() {
+        let test_program = test_program();
+        let mut vm: VirtualMachine<u8> = VirtualMachine::new(&test_program, None, false);
+        let mut cursor = std::io::Cursor::new(vec![1, 2, 3]);
+
+        let result = vm.read_value(&mut cursor);
+
+        assert!(result.is_ok());
+        assert_eq!(vm.cells[0], 1);
+    }
+
+    // does reading error correctly when it should?
+    #[test]
+    fn test_read_bad() {
+        let test_program = test_program();
+        let bad_instruction = test_program.instructions()[1].clone();
+        let mut vm: VirtualMachine<u8> = VirtualMachine::new(&test_program, None, false);
+        let mut cursor = std::io::Cursor::new([0; 0]);
+
+        let result = vm.read_value(&mut cursor);
+
+        assert!(result.is_err());
+    }
+
+    // does reading a byte into a cell work?
+    #[test]
+    fn test_write() {
+        let test_program = test_program();
+        let mut vm: VirtualMachine<u8> = VirtualMachine::new(&test_program, None, false);
+        let mut buf: Vec<u8> = vec![0; 10];
+        let mut cursor = std::io::Cursor::new(buf);
+
+        vm.cells[0] = 65;
+
+        let result = vm.print_value(&mut cursor);
+
+        assert!(result.is_ok());
+        assert_eq!(cursor.get_ref()[0], 65);
+    }
+
+    // does reading error correctly when it should?
```

Comment:
Again, surely `writing` and in this case the `when it should` could be better explained:

```suggestion
    // does writing error correctly when the output stream has no space?
```